app/controllers/users/totp_setup_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Users
+  class TotpSetupController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    # GET /users/totp
+    def index
+      if session[:otp_secret_key].nil?
+        session[:otp_secret_key] = current_user.generate_totp_secret
+      end
+      @qrcode = UserDecorator.new(current_user).qrcode(session[:otp_secret_key])
+    end
+
+    # PATCH /users/totp
```

Comment:
I don't have strong feelings about this, but perhaps lean towards removing them? It's one less thing to maintain in case they change. The canonical source is available at  `rake routes`, which is what I use to see what's going on, so I don't rely on these comments.